src/Store.js

```diff
@@ -176,122 +239,131 @@ class Store {
+      )
+    }
 
     if (fromServer === true) {
-      // If fromServer is true always fetch the data and return
-      return this.fetchOne(type, id, queryParams)
+      return this.fetchOne(type, id, options)
     } else if (fromServer === false) {
-      // If fromServer is false never fetch the data and return
-      return this.getRecord(type, id, queryParams)
+      return this.getOne(type, id, options)
+    }
+
+    const record = this.getOne(type, id, options)
+    if (record?.id) {
+      return record
     } else {
-      return this.findOrFetchOne(type, id, queryParams)
+      return this.fetchOne(type, id, options)
     }
   }
 
   /**
-   * returns cache if exists, returns promise if not
+   * Get all records with the given `type` and `ids` from the store. This will never fetch from the server.
    *
-   * @method findOrFetchOne
-   * @param {String} type record type
-   * @param id
-   * @param {Object} queryParams will inform whether to return cached or fetch
+   * @method getMany
+   * @param {String} type the type to get
+   * @param {String} ids the ids of the records to get
+   * @param {Object} options { queryParams }
+   * @return {Array} array of records
    */
-  findOrFetchOne = (type, id, queryParams) => {
-    // Get the matching record
-    const record = this.getMatchingRecord(type, id, queryParams)
+  getMany = (type, ids, options = {}) => {
+    const idsToQuery = ids.slice().map(String)
+    const records = this.getAll(type, options)
 
-    // If the cached record is present
-    if (record && record.id) {
-      // Return data
-      return record
-    } else {
-      // Otherwise fetch it from the server
-      return this.fetchOne(type, id, queryParams)
-    }
+    return records.filter((record) => idsToQuery.includes(record.id))
   }
 
   /**
-   * Like 'findOne', but with an array ids. If there are instances available in the store,
-   * it will return those, otherwise it will trigger a fetch
+   * Fetch all records with the given `type` and `ids` from the server.
+   *
+   * @method fetchMany
+   * @param {String} type the type to get
+   * @param {String} ids the ids of the records to get
+   * @param {Object} options { queryParams }
+   * @return {Promise} Promise.resolve(records) or Promise.reject(status)
+   */
+  fetchMany = (type, ids, options = {}) => {
+    let idsToQuery = ids.slice().map(String)
+    const queryParams = options.queryParams || {}
+    queryParams.filter = queryParams.filter || {}
+
+    const baseUrl = this.fetchUrl(type, queryParams)
+    const idQueries = deriveIdQueryStrings(idsToQuery, baseUrl)
+    const queries = idQueries.map((queryIds) => {
+      queryParams.filter.ids = queryIds
+      return this.fetchAll(type, { queryParams })
+    })
+    return Promise.all(queries)
+      .then((records) => [].concat(...records))
+      .catch((err) => Promise.reject(err))
+  }
```

Comment:
No reason not to do this - I just prefer then and catch with Promise.all since it's all confusing enough as is. Either way it's an async method. I checked in with Jon and he's okay with it.